### DISCUSSION TITLE
Path configuration DB

### DISCUSSION THREAD

**Discussion by llunohodl** [bluenviron/mediamtx#727] [General]
Is there any way to save all path confiurations added from HTTP API?
I use this server as RTSP proxy for accessing to IP cameras in VPN from Internet.
In my case I need:
- Add path over API
- Change 'readerUser' / 'readerPassword' over API

Sometimes I see that 'source' parameter of path become null and I cold not change it over API. Sometimes I can not remove and then create new path with same name. When I restart server API works fine again, BUT all paths added from API lost. 

For fix this problem I create wrapper API with SQL database that can reload all paths on server restart.

It will be great to add config option or API method for save all config changes from API in yml file.

API method "update" for paths will be useful too. "Update" means - crate new path or clean and rewrite config (if path exist).

**Comment by aler9** [comment]
Hello,

> Is there any way to save all path confiurations added from HTTP API?

The server was created to be run as a stateless service. Persistence is not provided internally since:

* databases are delicate and absolutely need redundance and high-availability; it would require too much effort to add such feature, given the fact that persistence can be added by using a wrapper, exactly like you do
* everyone has its own need, there isn't a DB solution that is good for all use cases. There are databases for speed (Redis), for indexing (Elasticsearch), for redundancy (etcd), general purpose (noSQLs, SQLs).

Maybe in the future the server will provide persistence by interacting with an external DB, for the moment, you can wrap it into a script and make it download the configuration from a safe place, exactly like most stateless services do.

> Sometimes I see that 'source' parameter of path become null and I cold not change it over API.

If you believe this is a bug, open an issue and describe how to replicate.

> API method "update" for paths will be useful too. "Update" means - crate new path or clean and rewrite config (if path exist).

paths are created from configurations. If you edit the configuration associated with a path, the path will be updated too.

[a comment by an automated account is not shown]
